lib/HttpClient/CurlClient.php

```diff
@@ -47,12 +47,21 @@ public function request($method, $absUrl, $headers, $params, $hasFile)
             throw new Error\Api("Unrecognized method $method");
         }
 
+        // Create a callback to capture HTTP headers for the response
+        $rheaders = array();
+        $headerCallback = function ($curl, $header_line) use (&$rheaders) {
+          list($key, $value) = explode(":", trim($header_line), 1);
+          $rheaders[trim($key)] = trim($value);
+          return strlen($header_line);
+        };
+
         $absUrl = Util\Util::utf8($absUrl);
         $opts[CURLOPT_URL] = $absUrl;
         $opts[CURLOPT_RETURNTRANSFER] = true;
         $opts[CURLOPT_CONNECTTIMEOUT] = 30;
         $opts[CURLOPT_TIMEOUT] = 80;
         $opts[CURLOPT_RETURNTRANSFER] = true;
+        $opts[CURLOPT_HEADERFUNCTION] = $headerCallback;
```

Comment:
Hmmm...emotionally caught between this option and http://stackoverflow.com/a/9183276/2515035 (split on `\r\n\r\n`). Any reason why you chose this option?